cve_bin_tool/checkers/perl.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+
+"""
+CVE checker for Perl
+References:
+https://www.cvedetails.com/vulnerability-list/vendor_id-1885/Perl.html
+"""
+from ..util import regex_find
+
+
+def get_version(lines, filename):
+    """
+    returns version information for perl as found in a given file.
+    """
+    regex = [r"Perl: ([0-9]+\.[0-9]+\.[0-9]+)"]
+    version_info = dict()
+    if filename.startswith("perl"):
+        version_info["is_or_contains"] = "is"
+
```

Comment:
This doesn't do any sort of search to find matches other than the filename, which is why it isn't returning any results and your tests are failing.  Take a look at some of the other checkers'  guess_contains* functions to get an idea, or re-use the same strng.  Do not look at curl or bluez, they're both bad at this.